_posts/2016-10-24-deploy-jekyll-with-travis.md

```diff
@@ -0,0 +1,232 @@
+
+One well-known (and completely understandable) limitation of this workflow: any Jekyll plugins not [whitelisted by GitHub Pages](https://pages.github.com/versions/) will be disabled when GitHub Pages builds the site, since `jekyll build` is run with the `--safe` tag. Many solutions exist to get around this, and when we decided to start using Jekyll plugins on our site we opted to use [Travis CI](https://travis-ci.org/) to build our site and push the compiled site to our `master` branch, to then be launched via GitHub Pages. To complicate matters, we recently starting building our site with gulp, which introduced another layer of dependencies to the mix.
+
+## Basic Workflow
+
+Our site was already using Travis CI to run our test suite on pull requests and
+merges to the `master` branch. You can learn about setting up Travis in their
+[docs](https://docs.travis-ci.com/).
+
+Previously, Travis would build the site and run our tests, then we would merge
+branches into `master`, triggering GitHub Pages to rebuild and deploy our site.
+But we want to build our site before it gets to GitHub Pages to ensure our
+Jekyll Plugins are included. In our new workflow:
+
+1. We'll create a new default branch so the `master` branch can be cleared
+out and used by Travis.
+2. When this new branch is updated (i.e. a pull request is merged), Travis will
+build the site.
+3. Using a GitHub token for authorization, Travis will commit the `_site`
+directory to the `master` branch, then push the `master` branch.
+4. This will trigger GitHub Pages to deploy our pre-built site.
+
+### Create a new default branch
+
+We created a new branch off `master` called `source` and set it to the default
+branch on GitHub. Each time the deploy process is run, we'll have Travis delete
+the files on the `master` branch, then commit the entire compiled `_site`
+directory. This will ensure that all changes (including deletions) are captured.
+
+### Set up a GitHub token for Travis
+
+To give Travis the authority to commit and push to our repository, we generated
+a personal access token by visiting the
+[settings page](https://github.com/settings/tokens) on GitHub and generating a
+new token. We used a bot account that has access to our repo to avoid needing to
+use one of our personal accounts.
+
+![Generating a personal access token in GitHub]({{ site.url }}/assets/img/blog/github-token.jpg)
+
+We'll need to use the access token and the account's email address in our
+deployment script, so to keep those items out of version control we used
+`travis encrypt`:
```

Comment:
I think it would be helpful to link to documentation on this. I needed to Google it to understand that better and found this: https://docs.travis-ci.com/user/encryption-keys/